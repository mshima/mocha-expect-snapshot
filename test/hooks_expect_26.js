const proxyquire = require('proxyquire');
const expect = require('expect26');
const jestSnapshot = require('jest-snapshot26');

const { setup, teardown } = proxyquire('../', { expect: expect, 'jest-snapshot': jestSnapshot });

describe('hooks with expect@26', function () {
  beforeEach(setup);

  afterEach(teardown);

  describe('toMachSnapshot()', function () {
    it('should match snapshot', function () {
      expect({ foo: 'bar' }).toMatchSnapshot();
    });
  });

  describe('toMachSnapshot() failure', function () {
    before(function () {
      this.snapshotResolverOptions = { rootDir: 'test2' };
      this.snapshotStateOptions = { updateSnapshot: false };
    });
    it('should not match snapshot', function () {
      expect(() => expect({ foo: 'bar' }).toMatchSnapshot()).toThrow();
    });
  });
});
