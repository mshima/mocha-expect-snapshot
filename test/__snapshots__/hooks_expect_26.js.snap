// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hooks with expect@26 toMachSnapshot() should match snapshot 1`] = `
Object {
  "foo": "bar",
}
`;
